smart-agent-did-resolver-api/initializers/smart-agent-did-resolver-api-initializers.js

```diff
@@ -0,0 +1,82 @@
+'use strict'
+
+const {
+  api,
+  Initializer
+} = require('actionhero')
+
+const {
+  DidResolver,
+  ContractLoader,
+  SignerInternal,
+  SignerIdentity
+} = require('@evan.network/api-blockchain-core')
+
+module.exports = class SmartAgentDidResolverApiInitializer extends Initializer {
+  constructor () {
+    super()
+    this.name = 'did-resolver-api'
+    this.loadPriority = 4100
+    this.startPriority = 4100
+    this.stopPriority = 4100
+  }
+
+  async initialize () {
+    if (api.config.smartAgentDidResolverApi.disabled) {
+      return
+    }
+
+    // specialize from blockchain smart agent library
+    class SmartAgentDidResolverApi extends api.smartAgents.SmartAgent {
+      async initialize () {
+        await super.initialize()
+      }
+
+      async resolveDid(did) {
+        const signerIdentity = this.retrieveSignerIdentity()
+        const resolver = new DidResolver({
+          ...this.runtime, signerIdentity
+        })
+        return resolver.getDidDocument(did)
+      }
+
+      async retrieveSignerIdentity() {
+        const web3 = this.runtime.web3
+
+        const contracts = await this.runtime.contracts;
+        const contractLoader =  new ContractLoader({
+          contracts,
+          web3,
+        });
+        const accountStore = this.runtime.accountStore;
+        const verifications = this.runtime.verifications;
+        const underlyingSigner = new SignerInternal({
+          accountStore,
+          contractLoader,
+          config: {},
+          web3,
+        });
+        const signer = new SignerIdentity({
```

Comment:
```suggestion
        const signer = new SignerIdentity(
          {
```

The objects that are passed into the SignerIdentity are wrapped using line breaks. I would suggest to also add a line break for the first input object.

Reply:
@ https://github.com/evannetwork/edge-server-did-resolver/pull/2#discussion_r355311504

current default naming schema is like @Philser used it:
`api/smart-agents/${NAME_OF_SMART_AGENT}` (and as name usually includes "smart-agent", this part is in the path twice.

Reply:
not quite sure of this ... needs another indentation ... I would rather put the "sub objects" also in one line like `}, {`

Reply:
another hint that we REALLY need a common linter config and code styles where we can commit to ;)